### ISSUE TITLE
Speedup for pattanaik

### ISSUE THREAD

**Issue by heckflosse** [LuminanceHDR/LuminanceHDR#86]
I created a new [branch](https://github.com/heckflosse/LuminanceHDR/tree/pattanaik_speeduo) in my fork with a speedup for pattanaik.

I used a 24 Mpixel file for my tests. Here are the timings:
```
                                  before                    after
local tonemapping off            60600 ms                  1480 ms
local tonemapping on            299300 ms                  6470 ms
```

Edit: There is still some potential for further speedups by better usage of vectorized code. But I've been lazy...

**Comment by Benitoite** [comment]
Nice work, that's five so far.  Fork to fork merge built for mac: https://filebin.net/uarumuwfxhbv3jxu/Luminance_HDR_2.5.2-SNAPSHOT-Test1e.dmg

**Comment by heckflosse** [comment]
Fixed with 0b394f7
